client/components/IndividualChallenge.js

```diff
@@ -1,107 +1,85 @@
-import React, { Component } from 'react';
+import React, { useState, useEffect } from 'react';
 import CodeMirror from 'react-codemirror';
 import { connect } from 'react-redux';
 import { putUserchallenge } from '../store/userchallenge';
 
-class _IndividualChallenge extends Component {
-  constructor() {
-    super();
-    this.state = {
-      html: '',
-      css: '',
-      js: '',
-    };
-    this.updateCodeHTML = this.updateCodeHTML.bind(this);
-    this.updateCodeCSS = this.updateCodeCSS.bind(this);
-    this.updateCodeJS = this.updateCodeJS.bind(this);
-    this.changeValue = this.changeValue.bind(this);
-    this.putValue = this.putValue.bind(this);
-  }
+const _IndividualChallenge = props => {
+  const [html, setHTML] = useState('');
+  const [css, setCSS] = useState('');
+  const [js, setJS] = useState('');
 
-  updateCodeHTML(newCode) {
-    this.setState({
-      html: newCode,
-    });
-  }
-
-  updateCodeCSS(newCode) {
-    this.setState({
-      css: newCode,
-    });
-  }
-
-  updateCodeJS(newCode) {
-    this.setState({
-      js: newCode,
-    });
-  }
-
-  changeValue() {
+  const changeValue = () => {
     // TODO: axios post call to the backend
-    console.log(this.state);
-  }
+    console.log({ html, css, js });
+  };
 
-  putValue(isSubmit) {
-    const userAnswer = { ...this.state, submitted: true, challengeId: this.props.id };
-    this.props
-      .putValue(userAnswer, this.props.id, isSubmit)
+  const putValue = isSubmit => {
+    const userAnswer = { html, css, js, submitted: true, challengeId: props.id };
+    props
+      .putValue(userAnswer, props.id, isSubmit)
       .then(userchallenge => console.log(userchallenge))
       .catch(ex => console.log(ex));
-  }
+  };
 
-  render() {
-    const challenge = { name: 'challenge1', description: 'draw a circle', difficulty: 1 };
-    const options = {
-      lineNumbers: true,
-      mode: 'javascript',
-    };
-    return (
-      <div className="d-flex flex-column align-items-center">
-        <h1>{challenge.name}</h1>
-        <p>{challenge.description}</p>
-        <div>
-          <div className="row">
-            <div className="col">users page goes here</div>
-            <div className="col">our image goes here</div>
+  const challenge = { name: 'challenge1', description: 'draw a circle', difficulty: 1 };
```

Comment:
Haoyu hard-coded this instead of pulling from props.  Fine if you leave this.  I created an issue to correct it.

Reply:
Ok. will leave for now.